### DISCUSSION TITLE
How do I get server-side environment variables on the client side?

### DISCUSSION THREAD

**Discussion by dingzhou-beck** [coder/code-server#6773] [Q&A]
I am deploying through a docker container and want to control whether certain views on the client are loaded through an environment variable on the server side (such as source control on the active bar).

I notice that in `src\vs\base\common\process. ts`  a system environment is exported, but this is not valid on the web.

In addition to the network interface, is there any other faster way to get the value or environment variable passed by the server on the client side?

**Comment by code-asher** [comment]
I am not sure this is possible, at least not without doing some patching.

The easiest way to get access to environment variables is probably by writing an extension, and then from there possibly there are APIs you could use to control the views.

**Reply by dingzhou-beck** [reply]
Yes, I think the extension approach is concise and effective. But I'm not sure if I can control the loading of the code-server system view by extending it.

**Reply by code-asher** [reply]
Yeah that is the big question.  If not, I think you might have to modify VS Code's source code to make this work.

**Reply by dingzhou-beck** [reply]
Well, thank you for your help.
